go/services/adr/client.go

```diff
@@ -0,0 +1,539 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package adr
+
+import (
+	"context"
+	"log/slog"
+	"sync"
+	"time"
+
+	"github.com/Azure/iot-operations-sdks/go/protocol"
+	"github.com/Azure/iot-operations-sdks/go/services/adr/adrnamegen/adrbaseservice"
+	"github.com/Azure/iot-operations-sdks/go/services/adr/adrtypegen/aeptypeservice"
+)
+
+const (
+	aepNameTokenKey = "aepName"
+	defaultTimeout  = 10 * time.Second
+)
+
+type (
+	Asset                          = adrbaseservice.Asset
+	AssetEndpointProfile           = adrbaseservice.AssetEndpointProfile
+	NotificationResponse           = adrbaseservice.NotificationResponse
+	AssetStatus                    = adrbaseservice.AssetStatus
+	AssetEndpointProfileStatus     = adrbaseservice.AssetEndpointProfileStatus
+	DetectedAsset                  = adrbaseservice.DetectedAsset
+	DiscoveredAssetEndpointProfile = aeptypeservice.DiscoveredAssetEndpointProfile
+)
+
+// Error represents an error returned by the ADR service.
+type Error struct {
+	Message       string
+	PropertyName  string
+	PropertyValue string
+	Condition     interface{}
```

Comment:
nit: Modern Go tends towards `any` instead of `interface{}`. I could have sworn `mage ci` autofixes this, though...

Reply:
I ran `mage ci` and it made formatting fixes, not sure why it wasn't amended. Changed to any